### PR TITLE
Update plugin to return full error by default

### PR DESCRIPTION
**Pull request by Subhajit97**
# Example query results
<details>
  <summary>Results</summary>
  
```
Add example SQL query results here (please include the input queries as well)
```
</details>

### CODE REVIEW

File: microsoft365/errors.go [hunk 1]
```diff
@@ -32,7 +33,11 @@ func getErrorObject(err error) *RequestError {
 		}
 	}
 
-	return nil
+	// If the type of error is other than ODataError
+	// return the exact error
+	return &RequestError{
+		Message: err.Error(),
```
Comment (cbruno10) [inline microsoft365/errors.go:39]: Are we able to return other properties from the error, like its code, or do we only have the message consistently?

**Review by cbruno10 — CHANGES_REQUESTED** [review]
@Subhajit97 Please see review, thanks!

File: microsoft365/errors.go (hunk 1 above)
Comment (Subhajit97) [reply microsoft365/errors.go]: @cbruno10 Generally, the errors are of type `*odataerrors.ODataError` or `*azidentity.AuthenticationFailedError`.
https://github.com/microsoftgraph/msgraph-sdk-go#31-get-the-users-drive
But if the error type is different (not sure what are the other common error types), we need to cast the error to get the proper error code.
For now, we are returning the entire API error as the message.

File: microsoft365/errors.go [hunk 2]
```diff
@@ -32,6 +35,27 @@ func getErrorObject(err error) *RequestError {
 		}
 	}
 
+	switch err := err.(type) {
+	case *odataerrors.ODataError:
+		if terr := err.GetError(); terr != nil {
```
Comment (cbruno10) [inline microsoft365/errors.go:40]: If we don't need the `if` check, we can remove it

File: microsoft365/errors.go [hunk 3]
```diff
@@ -32,6 +35,27 @@ func getErrorObject(err error) *RequestError {
 		}
 	}
```
Comment (cbruno10) [inline microsoft365/errors.go:37]: Please remove duplicate code above

**Review by cbruno10 — CHANGES_REQUESTED** [review]
@Subhajit97 Please see review comments, thanks!

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**